doc/spec/manifest.md

```diff
@@ -0,0 +1,241 @@
+        `<digest-alg-name>`s are:
+
+            tarsum.v1+sha256
+            sha256
+            sha512
+
+        The `<hex-encoded-digest>` value must contain hexadecimal digits only
+        and should be long enough to encode the full digest produced by the
+        corresponding digest algorithm.
+
+        Optionally, the `digest` value may be an absolute reference to a digest
+        in another repository:
+
+            <repository>@<digest-alg-name>:<hex-encoded-digest>
+
+        A `digest` in this form is considered relative to the repository
+        explicitly specified in the `<repository>` section of the reference.
+        The `<repository>` value must match the format specified by the
+        [Namespace and Discovery Documentation](https://github.com/dmcgowan/distribution/blob/namespace-proposal/doc/spec/namespace.md#name-grammar).
+    
+    Any media type may be in the `mediaType` field as the application may
+    use any type of content. If the `mediaType` is
+    `application/vnd.docker.distribution.content.manifest.v2+json` then the
+    dependency is another content manifest as described by this document. The
+    dependencies from that manifest should also be considered dependencies of
+    this application.
+
+    In general, the ordering of these dependencies is significant as it may be
+    significant to the application. However, it is best for the application
+    data below to also designate what actions to perform on the dependencies.
+
+- **`application`** *object*
+
+    Fields are:
+    
+    - **`mediaType`** *string*
+    
+        The MIME type of the JSON object in the `data` field.
+    
+    - **`data`** *object*
+    
+        Abstract application data in any schema, as described by the
+        `mediaType` field.
+    
+    Application data may be *any* JSON object, allowing this manifest format,
+    along with the generic dependencies list, to support any type of 
+    application that can be represented as a JSON configuration and a
+    collection of content-addressable blobs of data.
+
+- **`meta`** *object*
+
+    Meta may be *any* JSON object, allowing content creators to annotate
+    their application manifest with any additional labels beyond those already
+    defined by this manifest format. 
+
+### Example Manifest
+
+```json
+{
+    "schemaVersion": 2,
+    "createdAt": "2015-03-04T17:52:02.841028307-08:00",
+    "maintainers": [
+        {
+            "name": "Phillip Edward Nuñez",
+            "email": "pnunez@example.com",
+            "url": "https://example.com/pnunez/"
+        }
+    ],
+    "description": "Application to Simulate Turbo Encabulators",
+    "keywords": [
+        "foo",
+        "bar",
+        "baz"
+    ],
+    "website": "https://foo.bar.baz.example.com",
+    "licenses": [
+        "Apache-2.0"
+    ],
+    "dependencies": [
+        {
+            "mediaType": "application/vnd.docker.distribution.content.manifest.v2+json",
+            "length": 1024,
```

Comment:
We were taking a cue from TUF on this one. Providing the length keeps the client from having to trust the length as returned from the provider of the content (could be untrusted mirror). The potential risk being untrusted providers can return arbitrarily large blobs of data to fill up the disk of the client and the client would not be aware the data is incorrect until it believes it has received all the data.